plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/edit.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';
+import { BlockEditProps } from '@wordpress/blocks';
+import { getSettingWithCoercion } from '@woocommerce/settings';
+import { isBoolean } from '@woocommerce/types';
+
+/**
+ * Internal dependencies
+ */
+import { useIsDescendentOfSingleProductBlock } from '../../atomic/blocks/product-elements/shared/use-is-descendent-of-single-product-block';
+import { AddToCartOptionsSettings } from './settings';
+import { INNER_BLOCKS_TEMPLATE } from './constants';
+export interface Attributes {
+	className?: string;
+	isDescendentOfSingleProductBlock: boolean;
+}
+
+export type FeaturesKeys = 'isBlockifiedAddToCart';
+
+export type FeaturesProps = {
+	[ key in FeaturesKeys ]?: boolean;
+};
+
+export type UpdateFeaturesType = ( key: FeaturesKeys, value: boolean ) => void;
+
+// Pick the value of the "blockify add to cart flag"
+const isBlockifiedAddToCart = getSettingWithCoercion(
+	'isBlockifiedAddToCart',
+	false,
+	isBoolean
+);
+
+const AddToCartOptionsEdit = ( props: BlockEditProps< Attributes > ) => {
+	const { setAttributes } = props;
+
+	const blockProps = useBlockProps( {
+		className: 'wc-block-add-to-cart-with-options-wrapper',
```

Comment:
What do you think about removing the `-wrapper` part? If the block is called _Add to Cart with Options_, I would keep the CSS class in sync.